tests/transformers/test_transformers_model_export.py

```diff
@@ -1389,7 +1389,49 @@ def test_text2text_generation_pipeline_with_inference_configs(
     assert dict_inference == inference
 
 
-def test_text2text_generation_pipeline_with_params(text2text_generation_pipeline, tmp_path):
+def test_text2text_generation_pipeline_with_inference_config_and_params(
+    text2text_generation_pipeline, tmp_path
+):
+    data = "muppet keyboard type"
+    inference_config = {
+        "top_k": 2,
+        "num_beams": 5,
+        "top_p": 0.85,
+        "repetition_penalty": 1.15,
+    }
+    parameters = {"top_k": 3, "max_length": 30, "temperature": 0.62}
+    generated_output = mlflow.transformers.generate_signature_output(
+        text2text_generation_pipeline, data
+    )
+    signature = infer_signature(
+        data,
+        generated_output,
+        parameters,
+    )
+
+    model_path = tmp_path / "model"
+    mlflow.transformers.save_model(
+        text2text_generation_pipeline,
+        path=model_path,
+        inference_config=inference_config,
+        signature=signature,
+    )
+    pyfunc_loaded = mlflow.pyfunc.load_model(model_path)
+
+    # inference_config and default params are all applied
+    res = pyfunc_loaded.predict(data)
+    applied_params = inference_config.copy()
+    applied_params.update(parameters)
+    res2 = pyfunc_loaded.predict(data, applied_params)
+    assert res == res2
+
+    assert res != pyfunc_loaded.predict(data, {"max_length": 10})
+
+    # Extra params are ignored
+    assert res == pyfunc_loaded.predict(data, {"extra_param": "extra_value"})
```

Comment:
This could just be changed to an assertion that a string is returned from a call with an invalid param in order to help avoid flakiness